libevmasm/GasMeter.cpp

```diff
@@ -182,7 +180,16 @@ GasMeter::GasConsumption GasMeter::estimateMax(AssemblyItem const& _item, bool _
 		case Instruction::EXP:
 			gas = GasCosts::expGas;
 			if (u256 const* value = classes.knownConstant(m_state->relativeStackElement(-1)))
```

Comment:
Wouldn't it be simpler to have
```
if (u256 const* value = classes.knownConstant(m_state->relativeStackElement(-1)))
  if (*value)
  {
    // Note: msb() counts from 0
    unsigned significant_byte_count =  (boost::multiprecision::msb(*value) + 1 + 7) / 8;
    gas += GasCosts::expByteGas(m_evmVersion) * significant_byte_count;
  }
```

Or maybe even with ``if (init; cond)`` :-):

```
if (
  u256 const* value = classes.knownConstant(m_state->relativeStackElement(-1)));
  value && *value
)
{
  // Note: msb() counts from 0
  unsigned significant_byte_count =  (boost::multiprecision::msb(*value) + 1 + 7) / 8;
  gas += GasCosts::expByteGas(m_evmVersion) * significant_byte_count;
}
```

Reply:
Feel free to take the PR over :)